node.go

```diff
@@ -87,12 +97,69 @@ func (node *emptyNode) isEmpty() bool {
 
 var _ ProofNode = (*node)(nil)
 
+func (node *node) GetHash() []byte {
+	parent, ok := node.parent.(*interiorNode)
+	if !ok {
+		return nil
+	}
+	switch parent.leftChild.(type) {
+	case *emptyNode:
+		if &parent.leftChild.(*emptyNode).node == node {
+			return parent.leftHash
+		}
+	case *userLeafNode:
+		if &parent.leftChild.(*userLeafNode).node == node {
+			return parent.leftHash
+		}
+	default:
+		return nil
+	}
+
+	return parent.rightHash
```

Comment:
Don't forget to change this to ```leftHash``` if you're changing the return values in the switch statement.